static/app/stores/organizationsStore.tsx

```diff
@@ -36,7 +36,19 @@ const storeConfig: OrganizationsStoreDefinition = {
   },
 
   onUpdate(org) {
-    this.add(org);
+    let match = false;
+    this.state.forEach((existing, idx) => {
+      if (existing.id === org.id) {
+        this.state[idx] = {...existing, ...org};
+        match = true;
+      }
+    });
+    if (!match) {
+      throw new Error(
+        'Cannot update an organization that is not in the OrganizationsStore'
+      );
+    }
+    this.trigger(this.state);
```

Comment:
I'm slightly worried about implications here.

But basically onUpdate is supposed to be able to take a partial Organization (which is what is passed to the action), but it was just passing the org to `add` which DOES support updating an org, but if the org doesn't exist it will add it as new, which requires it to be a full organization object.

I think if tests pass OK here i'm fine with this.